### ISSUE TITLE
Beginless range in where condition does not match behavior of endless range.

### ISSUE THREAD

**Issue by maxcal** [rails/rails#38777]
### Steps to reproduce
```
rails g model post
rails db:migrate
```
```RUBY
Post.where(id: ..3)
```

### Expected behavior
 It should create the same query as `Post.where(id: Float::INFINITY..3)` and be the inverse of `Post.where(id: 3..)`.

```SQL
SELECT "posts".* FROM "posts" WHERE "posts"."id" <= $1 LIMIT $2
```

### Actual behavior
It creates a BETWEEN nil and end query:
```SQL
SELECT "posts".* FROM "posts" WHERE "posts"."id" BETWEEN $1 AND $2 LIMIT $3
```

### System configuration
**Rails version**:
6.0.2.1

**Ruby version**:
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux]

**Database adapter**:
pg (1.2.2)

**Comment by jonathanhefner** [comment]
I believe this was fixed by #36696 (and follow-up #38170), but these changes have not been backported to 6.0.

**Comment by eugeneius** [comment]
I backported https://github.com/rails/rails/pull/36696 (https://github.com/rails/rails/commit/1c3a6e8f539adfab45d40b53a6aa0bfeaa2f497f) and https://github.com/rails/rails/pull/38170 (https://github.com/rails/rails/commit/ca5c934aba60e1b146dc55d74f3fb8c1e01dc0a7) to 6-0-stable, so this will be fixed in the next 6.0.x release. Thanks for tracking those down, @jonathanhefner!
